frontend/src/bundles/video_editor/components/timeline/timeline.tsx

```diff
@@ -0,0 +1,70 @@
+import { type DragEndEvent, type ItemDefinition, type Range, type ResizeEndEvent,type RowDefinition, TimelineContext  } from 'dnd-timeline';
+import { useCallback, useState } from 'react';
+
+import { TimelineView } from './subcomponents/timeline-view.js';
+
+type Properties = {
+	initialRange: Range;
+	initialRows: RowDefinition[];
+	initialItems: ItemDefinition[];
+  };
+
+  const Timeline: React.FC<Properties> = ({ initialRange, initialRows, initialItems }) => {
+
+	const [range, setRange] = useState(initialRange);
+	const [rows] = useState(initialRows);
```

Comment:
I guess we don't need useState here because it is not changed anywhere below. In TimelineView you can simply pass initialRows as rows: `<TimelineView items={items} rows={initialRows} /> `

Reply:
But what if the user wants to create more rows? Doesnt it need to be state for the component to rerender with the new row?

Reply:
To add or remove row parent component can simply pass another set of rows to Timeline component and when it receives new props it automatically rerenders, so there is no need to create rows state in Timeline.

Reply:
Thank you